templates/user/view/block/user-name.php

```diff
@@ -3,7 +3,12 @@
 defined( 'ABSPATH' ) || exit;
 
 $display = get_option( 'hp_user_enable_display' );
+$online_users = (array) get_transient( 'online_users' );
```

Comment:
Please pass online user IDs via the request context instead (e.g. hivepress()->request->get_context('online_users'), then logic of fetching IDs will be in the User component).

Avoid adding logic to templates. For example, if we change the online users implementation later then this template part overridden via a child theme on some websites will become broken, templates should ideally contain HTML with variables, with possible conditions based on request context or options. In this case $display variable is created because it's used 2 times, but it's just an option which defines the output.